### PR TITLE
Mem check

### PR DESCRIPTION
**Pull request by davidHysom**
adding function in check_mem_capacity() in utils/lbann_library to report on estimated memory usage needed to load JAG samples. Compiles, but execution is not yet tested.

### CODE REVIEW

File: src/utils/lbann_library.cpp [hunk 1]
```diff
@@ -317,4 +318,149 @@ void print_lbann_configuration(lbann_data::Model *pb_model, lbann_comm *comm, in
   std::cout << std::endl;
 }
 
+void check_mem_capacity(lbann_comm *comm, const std::string sample_list_file, size_t stride, size_t offset) {
```
Comment (naoyam) [inline src/utils/lbann_library.cpp]: Why is this function defined here? Isn't it specific to JAG?

File: src/utils/lbann_library.cpp [hunk 2]
```diff
@@ -26,6 +26,7 @@
 
 #include "lbann/utils/lbann_library.hpp"
 #include "lbann/callbacks/callback_checkpoint.hpp"
+#include "lbann/data_store/data_store_jag.hpp"
```
Comment (naoyam) [inline src/utils/lbann_library.cpp]: Why do you need to load a specific data reader in lbann_library?

File: src/utils/lbann_library.cpp (hunk 1 above)
Comment (davidHysom) [reply src/utils/lbann_library.cpp]: It's specific to conduit, and all data_stores for all readers will be using conduit in the future, so I thought it was general enough to put here. It probably makes more sense to put it in the data_store_jag (which will be renamed data_store_conduit) class.

**Comment by davidHysom** [comment]
merging now, because I need this to track down a bug for Sam

**Pull request merged**